src/leet_cc_fmt_plug.c

```diff
@@ -164,7 +164,9 @@ static char *prepare(char *split_fields[10], struct fmt_main *self)
 	cp = mem_alloc(strlen(split_fields[0]) + strlen(split_fields[1]) + 2);
 	sprintf(cp, "%s$%s", split_fields[0], split_fields[1]);
 	if (valid(cp, self)) {
-		return cp;
+		char *cipher = str_alloc_copy(cp);
+		MEM_FREE(cp);
+		return cipher;
```

Comment:
Was there anything wrong with just returning `cp` here? Even if so, I don't see the extra allocation and copying improving anything, well, except for maybe hiding the memory leak if we "whitelist" leaks caused by `mem_alloc_tiny` (IIRC, Jim introduced debugging code to optionally forcibly free all such memory at program exit, so that we can look into the rest of the leaks without being bothered by these).

Reply:
Oh, I thought `mem_alloc_tiny` is smarter. I thought this memory would be freed once hashes are loaded.

The change silences ASan. OTOH there are other formats with similar silenced code, so I guess it might be fixed altogether later.